R/compare.R

```diff
@@ -142,16 +142,12 @@ order_proxy <- function(proxy, direction = "asc", na_value = "largest") {
   }
 
   if (is.data.frame(proxy)) {
-    # Avoid empty data frames getting a `NULL` order
+    # Work around type-instability in `base::order()`
     if (vec_size(proxy) == 0L) {
       return(integer(0L))
     }
-
     args <- unname(proxy)
-    order_expr <- expr(
-      base::order(!!!args, decreasing = decreasing, na.last = na.last)
-    )
-    eval_bare(order_expr)
+    exec(base::order, !!!args, decreasing = decreasing, na.last = na.last)
```

Comment:
Maybe `"order"` to improve backtraces in case `order()` fails?

Reply:
Is the backtrace also going to be ugly since `!!!args` is being done?

Reply:
No because these are scalars. Also the function is most important because it helps to identify the source of a problem.

Reply:
(oops sorry these are not scalars)